### PR TITLE
CONTRIB/WIRESHARK: Add IB Management Datagram RMPP LUA dissectors

### PR DESCRIPTION
**Pull request by tvegas1**
## What
Add RMPP dissector and register it to trigger on vendor specific payloads.

## Why ?
It useful to be able to decode perftest management datagram traffic.

## How ?

Tested on Wireshark, Windows. Run:
```
ibdump -o /tmp/a.pcap
wireshark -Xlua_script:rmpp.lua /tmp/a.pcap
```
or copy the LUA file in the Wireshark plugins directory:
- `Help` -> `About Wireshark` -> `Folders` -> `Personal Lua plugins`

### CODE REVIEW

File: contrib/wireshark/rmpp.lua [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    [RMPP_FLAG_FIRST]  = "First",
+    [RMPP_FLAG_LAST]   = "Last",
+}
+
+local rmpp_flag_set = {
+    [0] = "Not Set",
+    [1] = "Set",
+}
+
+local rmpp_proto = Proto("RMPP", "Reliable Message Passing Protocol")
```
Comment (brminich) [inline contrib/wireshark/rmpp.lua:81]: isn't it `Reliable  Multi-Packet Transaction Protocol`

File: contrib/wireshark/rmpp.lua [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+--   - Help -> About Wireshark -> Folders -> Personal Lua plugins
+
+local rmpp_version_str = {
+    [0] = "Not Set",
+    [1] = "1",
+}
+
+local RMPP_STATUS_SUCCESS    =   0
+local RMPP_STATUS_RESX       =   1
+local RMPP_STATUS_ABORT_MIN  = 118
```
Comment (brminich) [inline contrib/wireshark/rmpp.lua:19]: why is this definition needed?

File: contrib/wireshark/rmpp.lua [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+    [RMPP_STATUS_TMR]        = "TMR",
+    [RMPP_STATUS_UNSPEC]     = "Unspec",
+}
+
+local RMPP_TYPE_DATA  = 1
+local RMPP_TYPE_ACK   = 2
+local RMPP_TYPE_STOP  = 3
+local RMPP_TYPE_ABORT = 4
+
+local rmpp_valid_type = {
```
Comment (brminich) [inline contrib/wireshark/rmpp.lua:52]: how is it used?

Comment (tvegas1) [reply contrib/wireshark/rmpp.lua:52]: removed

File: contrib/wireshark/rmpp.lua (hunk 2 above)
Comment (tvegas1) [reply contrib/wireshark/rmpp.lua:19]: removed thanks. it is actually used by umad kernel module along with abort max to identifying failure code ranges

File: contrib/wireshark/rmpp.lua (hunk 1 above)
Comment (tvegas1) [reply contrib/wireshark/rmpp.lua:81]: fixed, seems that there is another rmpp that has some similarities while not being directly related to infiniband.

**Review by yosefe — DISMISSED** [review]

**Review by yosefe — APPROVED** [review]

**Pull request merged**